src/actions/ScanActions.js

```diff
@@ -10,21 +10,147 @@ import { selectWalletForExchange } from '../actions/CryptoExchangeActions.js'
+    // Verify if the user can satisfy the request
+    try {
+      // Check if the user owns a wallet with a matching native currency
+      const requestNativeAssets = assets.map(asset => asset.nativeCode)
+      const walletNativeAssets = Object.keys(currencyWallets).map(walletId => currencyWallets[walletId].currencyInfo.currencyCode)
+      if (!walletNativeAssets.some(walletAsset => requestNativeAssets.some(requestAsset => requestAsset.toUpperCase() === walletAsset.toUpperCase())))
+        throw new Error('No wallets found that support the native currencies listed in the request for payment address')
+
+      // Check if the user has enabled a specific token
+      let isTokenEnabled = false
+      for (const asset of assets) {
+        const supportingWalletIds = await getAssetSupportingWalletIds(currencyWallets, asset.nativeCode, asset.tokenCode)
+        if (supportingWalletIds.length > 0) isTokenEnabled = true
+      }
+      if (!isTokenEnabled)
+        throw new Error(
+          `No wallets found that have enabled a token listed in the request for payment address. Enable the token by long pressing the wallet and selecting 'Add / Edit Tokens'`
+        )
+    } catch (e) {
+      showError(e.message)
+    }
+
+    // Show wallet picker(s)
+    const jsonPayloadMap: { [currencyAndTokenCode: string]: string } = {}
+    for (const asset of assets) {
+      const reqNativeCode = asset.nativeCode.toUpperCase()
+      const excludeWalletIds = Object.keys(currencyWallets).filter(
+        walletId => currencyWallets[walletId].currencyInfo.currencyCode.toUpperCase() !== reqNativeCode
+      )
+      const reqTokenCode = asset.tokenCode != null ? asset.tokenCode.toUpperCase() : asset.nativeCode.toUpperCase()
+      const allowedCurrencyCode: string[] = [reqTokenCode]
+      await Airship.show(bridge => (
+        <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} excludeWalletIds={excludeWalletIds} allowedCurrencyCodes={allowedCurrencyCode} />
+      )).then(async ({ walletId, currencyCode }) => {
+        if (walletId != null && currencyCode != null) {
+          const wallet = currencyWallets[walletId]
+          const { publicAddress } = await wallet.getReceiveAddress({ currencyCode })
+          jsonPayloadMap[`${currencyWallets[walletId].currencyInfo.currencyCode}_${currencyCode}`] = publicAddress
+        }
+      })
+    }
+
+    // Handle POST or redirect
+    if (Object.keys(jsonPayloadMap).length === 0) {
+      showError('No wallets selected for request for payment address')
+    } else {
+      if (redir != null && redir !== '') {
+        // Make sure this isn't some malicious link to cause an infinite redir loop
+        const deepLink = parseDeepLink(redir)
+        if (deepLink.type === 'requestAddress' && deepLink.redir != null) throw new Error(`Invalid 'redir' query in request for payment address`)
+
+        // handle like any other deeplink
+        dispatch(launchDeepLink(parseDeepLink(redir)))
```

Comment:
This link should bounce out of Edge. launchDeepLink handles links internally in Edge but the most likely use case is to deep link into another app or to a website.